src/main/java/xyz/mkotb/configapi/internal/adapt/AdapterHandler.java

```diff
@@ -87,7 +87,8 @@ public static AdapterHandler create(NamingStrategy strategy) {
     }
     
     public static <I> void registerAdapter(Class<I> clazz, ObjectAdapter<I, ?> adapter) {
-        ADAPTERS.replace(clazz, adapter);
+        if (!ADAPTERS.replace(clazz, adapter))
+            ADAPTERS.add(clazz, adapter);
```

Comment:
Thanks! I think in this case, to support new adapters, it's probably best to just use the `put` method directly as it should perform this exact behaviour.